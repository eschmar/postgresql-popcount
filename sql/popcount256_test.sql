CREATE EXTENSION popcount;

SELECT popcount256(15::bit(4));
SELECT popcount256(7::bit(4));

SELECT popcount256(15::bit(8));
SELECT popcount256(7::bit(8));

SELECT popcount256(15::bit(12));
SELECT popcount256(7::bit(12));

SELECT popcount256(15::bit(32));
SELECT popcount256(7::bit(32));

SELECT popcount256(30::bit(32));
SELECT popcount256(127::bit(32));

SELECT popcount256(15::bit(256));
SELECT popcount256(7::bit(256));

SELECT popcount256(15::bit(512));
SELECT popcount256(7::bit(512));

SELECT popcount256(15::bit(8192));
SELECT popcount256(7::bit(8192));

SELECT popcount256(15::bit(1234));
SELECT popcount256(7::bit(1234));

SELECT popcount256(15::bit(3000));
SELECT popcount256(7::bit(3000));

SELECT popcount256(15::bit(8192));
SELECT popcount256(7::bit(8192));

SELECT sum(popcount256(x::bit(8192))) from generate_series(1,1000000) x;
SELECT sum(popcount256(x::bit(8192))) from generate_series(1,1000000) x;
