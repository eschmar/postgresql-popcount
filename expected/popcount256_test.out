CREATE EXTENSION popcount;
ERROR:  extension "popcount" already exists
SELECT popcount256(15::bit(4));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(4));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(8));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(8));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(12));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(12));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(32));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(32));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(30::bit(32));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(127::bit(32));
 popcount256 
-------------
           7
(1 row)

SELECT popcount256(15::bit(256));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(256));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(512));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(512));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(8192));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(8192));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(1234));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(1234));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(3000));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(3000));
 popcount256 
-------------
           3
(1 row)

SELECT popcount256(15::bit(8192));
 popcount256 
-------------
           4
(1 row)

SELECT popcount256(7::bit(8192));
 popcount256 
-------------
           3
(1 row)

SELECT sum(popcount256(x::bit(8192))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(popcount256(x::bit(8192))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

